lib/chef/client.rb

```diff
@@ -64,7 +65,21 @@ class Chef
   # The main object in a Chef run. Preps a Chef::Node and Chef::RunContext,
   # syncs cookbooks if necessary, and triggers convergence.
   class Client
+    class KeyMigration
+      include Singleton
+      attr_accessor :key_migrated
+      attr_accessor :old_priv_key
+      def initialize
+        @key_migrated = false
+        @old_priv_key = nil
```

Comment:
That is stubbed in for work I am doing now to update the cert on Chef Server AFTER certs are stored in the Certificate Store on the node AND when I need to update that cert because it is about to expire